CMakeLists.txt

```diff
@@ -1217,17 +1217,26 @@ else ()
   set(HAVE_HALF_FLOAT OFF)
 endif ()
 
+set(BUILD_PT_TRACER OFF)
+set(BUILD_PT_POST_PROCESSOR OFF)
 set(proc_supports_pt OFF)
+# Right now we only build PT related libraries on Linux x86_64.
 if (LINUX AND X86 AND X64)
-  # Right now we only build libipt on Linux x86_64.
-  # We define a variable to indicate whether DR will build libipt.
-  # TODO i#5505: In the next PR, we need to check whether the libipt submodule is initialized
-  # and print a warning message when the libipt directory is empty.
-  set(HAVE_LIBIPT ON)
+  # We will always build the PT tracer no matter whether the system supports PT.
+  set(BUILD_PT_TRACER ON)
+  # Check if the building machine support Intel PT. This function only checks if PT-related tests
+  # need to be built. PT-capable binaries can be built on any system.
   check_intel_pt_support(proc_supports_pt)
-else ()
-  set(HAVE_LIBIPT OFF)
-endif ()
+  # libdrpt2ir and drpt2trace depend on libipt. If the libipt submodule is not initialized,
+  # cmake will report an warning here.
+  if (EXISTS "${PROJECT_SOURCE_DIR}/third_party/libipt/.git" )
+    set(BUILD_PT_POST_PROCESSOR ON)
+  else ()
+    message(STATUS "libipt submodule is not initialized. libdrpt2ir and drpt2trace will not be built.")
+  endif ()
+else (LINUX AND X86 AND X64)
+  message(STATUS "PT related libraries only supported on Linux x86_64")
```

Comment:
In this case, some DR dox will contain broken references right?

Reply:
Again, we can separate building the code from building the docs, which is what we already do for extension libraries and platform-specific functions.

Reply:
Will do it in next PR.

Reply:
> In this case, some DR dox will contain broken references right?

This is a bug in my code's comments. I have fixed it. Sorry forgot to sync with you.